06_puppy/jimbotech/storer_test.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type storesSuite struct {
+	suite.Suite
+	store  Storer
+	mapper mapTest
+}
+
+func TestSuite(t *testing.T) {
+	var ms Storer = NewMapStore()
+	var sms Storer = &SyncMapStore{}
+	suite.Run(t, &storesSuite{store: ms, mapper: ms.(mapTest)})
```

Comment:
You could make this `suite.Run(t, &storesSuite{store: NewMapStore()})` - you don't need the var because you don't need to initialise the `mapper` field as that is initialised by the `SetupTest` function. `SetupTest` just needs the `store` field to be initialised. Just a thought - no need to make this change:

```
func TestSuite(t *testing.T) {
    suite.Run(t, &storesSuite{store: NewMapStore()})
    suite.Run(t, &storesSuite{store: &SyncMapStore{}})
}
```

I find shorter functions with no extraneous guff easier to read.